mycroft/skills/skill_manager.py

```diff
@@ -143,11 +143,18 @@ def run(self):
         # Scan the file folder that contains Skills.  If a Skill is updated,
         # unload the existing version from memory and reload from the disk.
         while not self._stop_event.is_set():
-            self._reload_modified_skills()
-            self._load_new_skills()
-            self._unload_removed_skills()
-            self._update_skills()
-            sleep(2)  # Pause briefly before beginning next scan
+            try:
+                self._reload_modified_skills()
+                self._load_new_skills()
+                self._unload_removed_skills()
+                self._update_skills()
+                sleep(2)  # Pause briefly before beginning next scan
+            except Exception as e:
+                LOG.exception('Something really unexpected has occured '
+                              'and the skill manager loop safety harness was '
+                              'hit.\n'
+                              'Exception: {}'.format(repr(e)))
```

Comment:
The log.exception method will automatically log the exception traceback.  No need to include the last line of the log message.

Reply:
You are technically correct, the best kind of correct!

Reply:
:+1: